control/pid.go

```diff
@@ -97,21 +97,49 @@ func (p *basicPID) reset() error {
 	if len(p.cfg.DependsOn) != 1 {
 		return errors.Errorf("pid block %s should have 1 input got %d", p.cfg.Name, len(p.cfg.DependsOn))
 	}
-	p.kI = p.cfg.Attribute.Float64("kI", 0.0)
-	p.kD = p.cfg.Attribute.Float64("kD", 0.0)
-	p.kP = p.cfg.Attribute.Float64("kP", 0.0)
-	p.satLimUp = p.cfg.Attribute.Float64("int_sat_lim_up", 255.0)
-	p.limUp = p.cfg.Attribute.Float64("limit_up", 255.0)
-	p.satLimLo = p.cfg.Attribute.Float64("int_sat_lim_lo", 0)
-	p.limLo = p.cfg.Attribute.Float64("limit_lo", 0)
+	p.kI = p.cfg.Attribute["kI"].(float64)
+	p.kD = p.cfg.Attribute["kD"].(float64)
+	p.kP = p.cfg.Attribute["kP"].(float64)
+
+	if satLimUp, ok := p.cfg.Attribute["int_sat_lim_up"].(float64); ok {
+		p.satLimUp = satLimUp
+	}
+
+	if limup, ok := p.cfg.Attribute["limit_up"].(float64); ok {
+		p.limUp = limup
+	}
+
+	if p.cfg.Attribute.Has("int_sat_lim_lo") {
+		p.satLimLo = p.cfg.Attribute["int_sat_lim_lo"].(float64)
```

Comment:
is there something different about this and the next that you aren't checking `ok` and using `.Has()` instead?

Reply:
No - nice catch I'll make them consistent.